segmentation/separate_leaves.py

```diff
@@ -36,17 +30,77 @@ def mask_leaf_parts(img):
     
     return masked
 
-#%% find bounding rectangles per leaf, save them as individual images
-contours = find_contours(img)
+#%%
+
+def get_bounding_boxes(path_masked):
+    # assumes image is masked
+    img_orig = cv2.imread(path_masked)
+    # plt.imshow(img_orig)
+    img = cv2.cvtColor(img_orig, cv2.COLOR_BGR2RGB)
+    # plt.imshow(img)
 
-for i, c in enumerate(contours):
-    x,y,w,h = cv2.boundingRect(c)
+    contours = find_contours(img)
+    # print(len(contours))
+    contours = contours[1:]
+
+    boxes = []
+
+    for i, c in enumerate(contours[::-1]):
+        if (cv2.contourArea(c) > 10000):
+            boxes.append(cv2.boundingRect(c))
+    
+    return boxes
+
+
+def cut(img, bounding_boxes, is_masked=True):
+    segments = []
+
+    for x,y,w,h in bounding_boxes:
+        img_segmented = img[y:y+h, x:x+w]
+        
+        if is_masked:
+            img_segmented = mask_leaf_parts(img_segmented)
+    
+        segments.append(img_segmented)
+    
+    return segments
+    
+
+def write(segments, path, img_original):
+    filename = re.findall(r'[^\/]+(?=\.)', path)[0]
```

Comment:
What does the regex do?

Reply:
finds the filename from path without the file extension